pages/ai-ecosystem/graph-rag/knowledge-graph-creation.mdx

```diff
@@ -0,0 +1,152 @@
+
+<Cards>
+  <Cards.Card
+    title="LlamaIndex"
+    href="/ai-ecosystem/integrations#llamaindex"
+  />
+    <Cards.Card
+    title="LangChain"
+    href="/ai-ecosystem/integrations#langchain"
+  />
+</Cards>
+
+For more information, refer to the following:
+
+- [Improved Knowledge Graph Creation with LangChain and LlamaIndex](https://memgraph.com/blog/improved-knowledge-graph-creation-langchain-llamaindex)
+- [Integrating Memgraph with LlamaIndex Building GenAI Apps](https://memgraph.com/blog/integrating-memgraph-llamaindex-genai-apps)
+- [Constructing a Knowledge Graph with LlamaIndex and Memgraph](https://www.llamaindex.ai/blog/constructing-a-knowledge-graph-with-llamaindex-and-memgraph)
+- [How to Build GenAI Apps with LlamaIndex and Memgraph](https://www.youtube.com/watch?v=CRc0yHsmCXA)
+
+#### Production-ready approach
+
+From our experience working with unstructured data, here’s what we recommend. For production-ready systems or when accuracy matters, you need a custom pipeline which may or may not look smtg like this:
+
+1. **Preprocessing** to clean and organize raw data for analysis.
+2. **Named Entity Recognition (NER)** to dentify and classify entities in the data.
+3. **Relationship extraction** to detect and define connections between entities.
+4. **Contextual understanding and disambiguation** to resolve ambiguities and ensure contextually accurate representations.
+5. **Post-processing** to refine and validate the structured data before graph creation.
+
+By following these steps, you can ensure your knowledge graph is precise, factual, and optimized for GraphRAG use cases.
+
+Here’s an example: [How to Extract Entities and Build a Knowledge Graph with Memgraph and SpaCy](https://memgraph.com/blog/extract-entities-build-knowledge-graph-memgraph-spacy). 
+
+## Knowledge graphs in Memgraph
+
+Memgraph provides an ideal platform for storing and querying knowledge graphs. You can even ingest data with **Kafka**, **Redpanda** or **Pulsar** and process updates on the fly.
+
+![streaming](/pages/ai-ecosystem/streaming.png)
+
+For structured data, refer to [Import best practices](https://memgraph.com/docs/data-migration/best-practices). 
+
+For unstructured data, depending on your use case and goal, create a pipeline to pre-process and transform it into graph-ready format.
+
+<Cards>
+  <Cards.Card
+    title="Read more about knowledge graphs"
+    href="/data-modeling/knowledge-graph"
+  />
+</Cards>
+
+## Next Steps
+
+Visit **Knowledge graph retrieval** and **Examples and demos** pages to see how knowledge graphs are applied in real-world scenarios.
+
+<Cards>
+  <Cards.Card
+    title="Knowledge graph retrieval"
+    href="/ai-ecosystem/graph-rag/knowledge-retrieval"
+  />
+    <Cards.Card
+    title="Examples and demos"
+    href="/ai-ecosystem/graph-rag/examples-and-demos"
+  />
+</Cards>
```

Comment:
Again, link in the text, not in separate cards. We add cards for additional content, not mentioned in the text.